cmd/geth/retesteth.go

```diff
@@ -201,12 +196,50 @@ func (api *RetestethAPI) SetChainParams(ctx context.Context, chainParams ChainPa
 }
 
 func (api *RetestethAPI) MineBlocks(ctx context.Context, number uint64) (bool, error) {
-	for i := 0; i < int(number); i++ {
-		if err := api.mineBlock(); err != nil {
-			return false, err
+	currentNumber := api.testclient.ethereum.BlockChain().CurrentBlock().Number().Uint64()
+	startBlock := currentNumber
+	upToBlock := currentNumber + number
+	api.testclient.ethereum.StartMining(1)
+	api.testclient.ethereum.Miner().SetRecommitInterval(time.Second * 10)
+	fakeBlockRemoved := false
+
+	for currentNumber < upToBlock {
+		currentNumber = api.testclient.ethereum.BlockChain().CurrentBlock().Number().Uint64()
+		if currentNumber == startBlock + 1 && fakeBlockRemoved == false {
+			api.testclient.ethereum.BlockChain().SetHead(startBlock)
+			currentNumber = startBlock
+			fakeBlockRemoved = true
 		}
 	}
-	fmt.Printf("Mined %d blocks\n", number)
+	api.testclient.ethereum.StopMining()
+	//time.Sleep(time.Second * 10)
+	api.testclient.ethereum.BlockChain().SetHead(upToBlock)
```

Comment:
It looks very brittle -- you're using SetHead a lot. SetHead is pretty 'unsafe' to use, it pretty heavily just messes with the internals and doesn't really make sure to clean everything up. It's a "use at your own risk" sort of tool which is generally used as a last resort

Reply:
if you are going to implement a state transition generation tool, that would be basically test_mineBlocks function. and with that I will be able to run retesteth on all geth native api. 
just need to sort out the consenus around `debug_accountRangeAt` and `debug_storageRangeAt` 

with that I would say I will be able to support retesteth.go and test generation on this state transition tool-only. however it will take some time to go through the process and see how tests are going to be generated with that transition tool.

*** 

besides the tool I mean, I am kin to the idea of the test API that will run small unit tests like we have RLP,Transaction,simpleVM, difficulty and others. as well as general rpc api implementation tests. (custom for the clients to implement) 

the  reason behind it is the idea to create a test development IDE that would simplify the process of test creation for the user. so it would not be required to edit json files manually, but just describe the test with a few simple lines like: 
```
test:           a = add(1, 1)  
condition:  a = 2
```
that would be translated into .json filler and then to blockchain/simple vm test. so to say. this just a long term idea or perhaps if I had more people in the team...